### PR TITLE
Add unit test for getlocal

### PR DESCRIPTION
**Pull request by BrianCurtis-NOAA**
* Make creategrib into module for use in writing tests
* Fix tests that previously used old creategrib
* Add new test for getlocal g2 function/call

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -68,11 +68,9 @@ foreach(kind ${kinds})
   target_link_libraries(${lib_name}_c PRIVATE ${PNG_LIBRARIES})
 
   add_library(${lib_name} STATIC $<TARGET_OBJECTS:${lib_name}_f> $<TARGET_OBJECTS:${lib_name}_c>)
-  if(bacio_VERSION GREATER_EQUAL 2.5.0)
-    target_link_libraries(${lib_name} PUBLIC bacio::bacio ${PNG_LIBRARIES} ${JASPER_LIBRARIES})
-  else()
-    target_link_libraries(${lib_name} PUBLIC bacio::bacio_4 ${PNG_LIBRARIES} ${JASPER_LIBRARIES})
-  endif()
+
+  target_link_libraries(${lib_name} PUBLIC bacio::bacio ${PNG_LIBRARIES} ${JASPER_LIBRARIES})
```
Comment (edwardhartnett) [inline src/CMakeLists.txt:72]: Why did you change the handling of bacio names?

File: tests/test_getlocal.f90 [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    csec2_comp = (/ achar(1), achar(2), achar(3) /)
+
+    if (ierr .ne. 0) stop 3
+    if (lcsec2 .ne. 3) stop 4
+
+    do i = 1, 3
+        if (csec2(i) .ne. csec2_comp(i)) stop 5
+    end do
+
```
Comment (edwardhartnett) [inline tests/test_getlocal.f90:27]: Is there anything else you can check in the GRIB message?

File: src/CMakeLists.txt (hunk 1 above)
Comment (BrianCurtis-NOAA) [reply src/CMakeLists.txt:72]: @edwardhartnett It's handled here: https://github.com/NOAA-EMC/NCEPLIBS-g2/blob/b05a08644ed8ce638ad579240c1f61df2625ce23/CMakeLists.txt#L29-L31

**Review by edwardhartnett — APPROVED** [review]

File: tests/test_getlocal.f90 (hunk 2 above)
Comment (BrianCurtis-NOAA) [reply tests/test_getlocal.f90:27]: I can test more of the code without going and changing the GRIB message itself.
